ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -880,13 +906,19 @@ def set_waypoints_velocity(self):
                     recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                                                self.lookahead_wps,
                                                dist_to_tl - (self.dyn_tl_buffer - 1.0))
+                    self.stop_target = self.next_tl_wp
             else:
                 # not sure when this might trigger but not match next if statement - then 
                 # reduce car velocity to 0.0 stoping before lights
                 # if self.state != 'slowdown' and self.state != 'stopped':
                 #    rospy.logwarn("set car state to slowdown at ptr = {}".format(self.final_waypoints_start_ptr))
 
                 if dist_to_tl - self.dyn_tl_buffer > self.min_stop_distance:
+                    if self.state != 'slowdown':
+                        rospy.logwarn("Within stopping distance of TL - Start to slowdown!")
+                        self.stop_target = self.next_tl_wp
+                    else:
+                        rospy.logwarn("we should not get here! {}, {}".format(self.stop_target, self.next_tl_wp))
                     recalc = self.produce_slowdown(self.final_waypoints_start_ptr,
                                                self.lookahead_wps,
                                                dist_to_tl - (self.dyn_tl_buffer - 1.0))
```

Comment:
For the warning on line 926 below this, isn't that the condition where the red light activates when the car is still outside of the creep_zone but min_stop_distance is too long to slowdown in time?  If so, maybe the warning should be something like "Distance to red light is too short to slow down in time, continue on."  Need to try to test this condition, but I'm not sure if the max speed is high enough to get it, or we could test with a shorter creep_zone.